### PR TITLE
[IMP] mail: message action registry + reorder actions in inbox

### PR DESCRIPTION
**Pull request by phenix-factory**
Before this commit, message actions were all defined in template
of `Message` component. It's good when there are few actions and
the ordering of actions is always the same, but there are now
many actions and we want to order these actions depending on context.

This commit fixes the issue by introducing a special registry to add
message actions. This registry is quite similar to thread actions:
each item define conditions to be shown, behaviour on click, and
their sequence which can depend on the context where the message is
being viewed.

In addition to this message action registry, this commit also
reorder message actions in inbox as follow:

1. Quick actions:
 - Add a Reaction
 - Mark as To do
 - Mark as Read
2. More menu:
 - View reactions
 - Reply
 - Unfollow
 - Edit
 - Delete

Task-3458086

### CODE REVIEW

**Comment by alexkuhn** [comment]
I would introduce `message_actions` with a similar architecture to `thread_actions`, so that it's much easier to adapt for new specs

File: addons/mail/static/src/core/common/message.xml [hunk 1]
```diff
@@ -135,22 +135,24 @@
...
-                    <DropdownItem t-if="props.thread.model === 'discuss.channel' and store.user" class="'px-2 d-flex align-items-center rounded-0'" title="'Mark as Unread'" onSelected="() => this.onClickMarkAsUnread()"><i class="fa fa-lg fa-fw fa-eye-slash pe-2"/>Mark as Unread</DropdownItem>
-                    <DropdownItem t-if="editable" class="'px-2 d-flex align-items-center rounded-0'" title="'Edit'" onSelected="() => this.onClickEdit()"><i class="fa fa-lg fa-fw fa-pencil pe-2"/>Edit</DropdownItem>
-                    <DropdownItem t-if="deletable" class="'px-2 d-flex align-items-center rounded-0'" title="'Delete'" onSelected="() => this.onClickDelete()"><i class="fa fa-lg fa-fw fa-trash pe-2"/>Delete</DropdownItem>
+
+                    <t t-foreach="messageActions.actions.filter(action => action.dropdown)" t-as="action" t-key="action.id">
+                        <DropdownItem class="'px-2 d-flex align-items-center rounded-0'" title="action.title" onSelected="action.onClick">
```
Comment (alexkuhn) [inline addons/mail/static/src/core/common/message.xml:151]: Would be nice to not explicitly tell whether it's in dropdown or not.
Like thread actions, this is automatically deduced by amount of actions and their sequences.
Let's say at most 4 actions immediately visible (including "Expand").

File: addons/mail/static/src/core/common/message.xml [hunk 2]
```diff
@@ -135,20 +135,20 @@
...
                 <button t-else="" class="btn px-1 py-0 rounded-0" t-att-title="action.title" t-att-name="action.id" t-on-click.stop="action.onClick">
                     <i class="fa fa-lg" t-att-class="action.icon"/>
                 </button>
             </t>
-            <Dropdown t-if="messageActions.actions.length gt messageActionLength" onStateChanged="state => this.state.expandOptions = state.open" position="props.message.eq(props.thread?.newestMessage)  ? 'top-start' : 'bottom-start'" togglerClass="`btn p-0 ${ state.expandOptions ? 'bg-200' : '' }`" menuClass="'d-flex flex-column py-0 o-mail-Message-moreMenu'" class="'d-flex rounded-0'" title="expandText">
+            <Dropdown t-if="messageActions.actions.length gt quickActionCount" onStateChanged="state => this.state.expandOptions = state.open" position="props.message.eq(props.thread?.newestMessage)  ? 'top-start' : 'bottom-start'" togglerClass="`btn p-0 rounded-0 ${ state.expandOptions ? 'bg-200' : '' }`" menuClass="'d-flex flex-column py-0 o-mail-Message-moreMenu'" class="'d-flex rounded-0'" title="expandText">
```
Comment (phenix-factory) [inline addons/mail/static/src/core/common/message.xml:144]: @alexkuhn Why is there diff on  `rounded-0`? There is already one on class.

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**